components/patient-data/rest/src/main/java/org/phenotips/data/rest/internal/DefaultPatientByExternalIdResourceImpl.java

```diff
@@ -201,4 +211,60 @@ private Response checkForMultipleRecords(Patient patient, String eid)
         }
         return null;
     }
+
+    /**
+     * Checks patient records for duplicates. Creates and updates a patient if no existing patients with {@code eid} are
+     * found, otherwise lists duplicate records in the response.
+     *
+     * @param eid the external ID of the patient of interest
+     * @param json the data to update the patient with
+     * @return a {@link Response} with no content if a new patient is created, a list of patients if duplicate patients
+     *         exist, or an error code in the event of error
+     */
+    private Response checkRecords(final String eid, final String json)
+    {
+        final Response response = checkForMultipleRecords(null, eid);
+        return (response != null && response.getStatus() == Status.NOT_FOUND.getStatusCode())
+            ? createPatient(eid, json)
+            : response;
+    }
+
+    /**
+     * Tries to create a patient with provided {@code eid} and {@code json data}. Returns a {@link Response} with
+     * no content if successful, an error code otherwise.
+     *
+     * @param eid the external identifier for the patient; will be overwritten if an eid is specified in {@code json}
+     * @param json patient data as json string
+     * @return a {@link Response} with no content if successful, an error code otherwise
+     */
+    private Response createPatient(final String eid, final String json)
+    {
+        this.logger.debug("Creating patient record with external ID [{}]", eid);
+        try {
+            if (StringUtils.isBlank(json)) {
+                this.logger.error("Provided patient json: {} is invalid.", json);
+                return Response.status(Status.BAD_REQUEST).build();
+            }
+            final JSONObject patientJson = new JSONObject(json);
+            final User currentUser = this.users.getCurrentUser();
+            if (!this.access.hasAccess(Right.EDIT, currentUser == null ? null : currentUser.getProfileDocument(),
+                this.currentResolver.resolve(Patient.DEFAULT_DATA_SPACE, EntityType.SPACE))) {
+                this.logger.error("Edit access denied to user [{}].", this.users.getCurrentUser());
+                return Response.status(Status.FORBIDDEN).build();
+            }
+            final Patient patient = this.repository.create();
+            // Since we're creating a new patient, if eid is not provided in patientJson, use the eid that was passed in
+            if (StringUtils.isBlank(patientJson.optString(EID_LABEL))) {
```

Comment:
For an existing patient, sending a blank eid will actually blank it. I think we should allow the same here. Replace with `json.has`.